platforms/core-runtime/base-services/src/main/java/org/gradle/internal/buildtree/BuildModelParameters.java

```diff
@@ -61,4 +61,10 @@ public interface BuildModelParameters {
      * This parameter is only used for benchmarking purposes.
      */
     boolean isModelAsProjectDependency();
+
+    /**
+     * Returns true if the model building is resilient so some failures in model building.
+     * @return true if the model building is resilient, false otherwise
+     */
+    boolean isResilientModelBuilding();
```

Comment:
🤔  This should probably be part of CC key, since switching this flag on/off can result in changing the build/sync result between failure and success. Formally, it does not affect CC, since today it's never used for model build on its own, but this affects IP, which relies on the CC key validation, when following the cache hit scenario.